oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/IndexName.java

```diff
@@ -231,26 +231,37 @@ public String nextCustomizedName() {
     }
 
     public static boolean isIndexActive(String indexPath, NodeState rootState) {
+        return isIndexActive(indexPath, rootState, 0);
+    }
+
+    private static boolean isIndexActive(String indexPath, NodeState rootState, int recursionDepth) {
         // An index is active if it has a hidden child node that starts with ":oak:mount-",
         // OR if it is an active merged index
-        try {
-            NodeState indexNode = rootState;
-            for (String e : PathUtils.elements(indexPath)) {
-                indexNode = indexNode.getChildNode(e);
+        if (recursionDepth > 50) {
+            LOG.warn("Fail to check index activeness for {} due to high recursion depth", indexPath);
+            return true;
+        }
+        NodeState indexNode = rootState;
+        for (String e : PathUtils.elements(indexPath)) {
+            indexNode = indexNode.getChildNode(e);
+        }
```

Comment:
This logic looks strange to me, it took me a while to understand it. It's trying to get the child node of the last element on the indexPath, right? Can you re-write this in a way that it is more clear? Something that reads more like `PathUtils.elements(indexPath).last().getChildNode(e)`?

Reply:
This is a _loop_ over all the entries in the path, to go down to the respective node. This code is not changed.